compiler/rustc_middle/src/ty/context.rs

```diff
@@ -2650,6 +2655,30 @@ impl<'tcx> TyCtxt<'tcx> {
         self.mk_const(ty::ConstS { kind: ty::ConstKind::Infer(ic), ty })
     }
 
+    #[inline]
+    pub fn mk_const(self, mut c: ty::ConstS<'tcx>) -> ty::Const<'tcx> {
```

Comment:
is that necessary? I would expect that having a param effect here shouldn't hurt anyone

Reply:
Ah this was my heavy-handed approach to eliminate occurrences of `cannot use constants which depend on generic parameters in types`. Could probably just ignore effects at that error site?

Reply:
yeah, ignoring effects there is probably the better fix